bentoml/_internal/frameworks/catboost.py

```diff
@@ -0,0 +1,273 @@
+
+    .. code-block:: python
+
+        import bentoml
+        # target model must be from the BentoML model store
+        booster = bentoml.catboost.load_model("my_catboost_model")
+    """  # noqa: LN001
+    if not isinstance(bento_model, bentoml.Model):
+        bento_model = get(bento_model)
+        assert isinstance(bento_model, bentoml.Model)
+
+    if bento_model.info.module not in (MODULE_NAME, __name__):
+        raise NotFound(
+            f"Model {bento_model.tag} was saved with module {bento_model.info.module}, not loading with {MODULE_NAME}."
+        )
+
+    model_file = bento_model.path_of(MODEL_FILENAME)
+    cb_class_name: str = bento_model.info.metadata.get(
+        MODEL_TRAINING_CLASS_NAME_KEY, DEFAULT_MODEL_TRAINING_CLASS_NAME
+    )
+    cb_class: t.Type[cb.CatBoost] = getattr(cb, cb_class_name, cb.CatBoost)
+    assert (
+        issubclass(cb_class, cb.CatBoost) or cb_class is cb.CatBoost
+    ), f"{cb_class_name} is not a valid CatBoost class."
+    cb_instance = cb_class()
+    booster: cb.CatBoost = cb_instance.load_model(fname=model_file)
+    return booster
+
+
+def save_model(
+    name: str,
+    model: cb.CatBoost,
+    *,
+    signatures: dict[str, ModelSignatureDict] | None = None,
+    labels: dict[str, str] | None = None,
+    custom_objects: dict[str, t.Any] | None = None,
+    metadata: dict[str, t.Any] | None = None,
+) -> bentoml.Model:
+    """
+    Save an CatBoost model instance to the BentoML model store.
+
+    Args:
+        name (``str``):
+            The name to give to the model in the BentoML store. This must be a valid
+            :obj:`~bentoml.Tag` name.
+        model (:obj:`~catboost.CatBoost`):
+            The CatBoost model to be saved.
+        signatures (``dict[str, ModelSignatureDict]``, optional):
+            Signatures of predict methods to be used. If not provided, the signatures default to
+            ``{"predict": {"batchable": False}}``. See :obj:`~bentoml.types.ModelSignature` for more
+            details.
+        labels (``dict[str, str]``, optional):
+            A default set of management labels to be associated with the model. An example is
+            ``{"training-set": "data-1"}``.
+        custom_objects (``dict[str, Any]``, optional):
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``.
+
+            Custom objects are currently serialized with cloudpickle, but this implementation is
+            subject to change.
+        metadata (``dict[str, Any]``, optional):
+            Metadata to be associated with the model. An example is ``{"max_depth": 2}``.
+
+            Metadata is intended for display in model management UI and therefore must be a default
+            Python type, such as ``str`` or ``int``.
+    Returns:
+        :obj:`~bentoml.Tag`: A :obj:`tag` with a format `name:version` where `name` is the
+        user-defined model's name, and a generated `version` by BentoML.
```

Comment:
```suggestion
    Args:
        name: The name to give to the model in the BentoML store. This must be a valid
                    :obj:`~bentoml.Tag` name.
        model: The CatBoost model to be saved.
        signatures: Signatures of predict methods to be used. If not provided, the signatures default to
                            ``{"predict": {"batchable": False}}``. See :obj:`~bentoml.types.ModelSignature` for more
                            details.
        labels: A default set of management labels to be associated with the model. An example is
                    ``{"training-set": "data-1"}``.
        custom_objects: Custom objects to be saved with the model. An example is
                                     ``{"my-normalizer": normalizer}``. Custom objects are currently serialized with 
                                     cloudpickle, but this implementation is subject to change.
        metadata: Metadata to be associated with the model. An example is ``{"max_depth": 2}``.
                          Metadata is intended for display in model management UI and therefore must be a default
                          Python type, such as ``str`` or ``int``.
 
    Returns:
        :obj:`~bentoml.Model`: A :obj:`bentoml.Model` object.
```

Reply:
since we already type hint the function, no need to do it in docs as sphinx.ext.napoleon will take care of this for us.

Reply:
It is nice to make it incrementally.